server/use_cases/add_service_commitments.py

```diff
@@ -46,3 +75,99 @@ def check_time_overlap(shifts):
                 shifts[i].shift_end > shifts[j].shift_start):
                 return True
     return False
+
+# test cases
+if __name__ == "__main__":
+    from unittest.mock import Mock
+
+    # valid shift ID (successful case)
+    def test_valid_shift_commitment():
+        shifts_repo = Mock()
+        commitments_repo = Mock()
+
+        valid_shift = Mock()
+        valid_shift.get_id.return_value = (
+            "mocked-valid-shift-id"
+        )
+        shifts_repo.get_shifts.return_value = [valid_shift]
+
+        valid_commitment = Mock()
+        valid_commitment.service_shift_id = "mocked-valid-shift-id"
+
+        valid_commitment.to_dict.return_value = (
+            {"_id": "commitment-id"}
+        )
```

Comment:
Commitment repo assigns the commitment-id when a commitment is added to the repo. When we send a commitment to the add_service_commitments use case, the commitments do not have an id. You'll want to modify your code such that commitments_repo returns ['commitment-id'] when insert_service_commitments is called.